### PR TITLE
feat: exclude re-exported functions from checkExportsAreDocumented()…

### PR DESCRIPTION
**Pull request by gladkia**
The goal of this PR is to exclude the man page for re-exported functions from `checkExportsAreDocumented()` check as it's not possible to add runnable examples for the re-exported functions.

If you are submitting a pull request to `BiocCheck` please follow the
[instructions](https://docs.google.com/presentation/d/1DkN2WVPOMVGqUtlSSrWbx6IMjtGP_cEHoE3nfOEnD68/edit#slide=id.p).
The presentation includes steps for forking, creating a working branch, and
useful related information.

For a successful merge, the following steps are required:

* [X] Update the NEWS file
* [X] Update the vignette file
* [X] Add unit tests (optional but highly recommended)
* [X] Passing `R CMD build` & `R CMD check` on Bioconductor devel

List a reviewer in the Pull Request (either @LiNk-NY, @lshep, @mtmorgan).
Reviewers will make sure to `Comment`, `Approve` or `Request changes`.
@LiNk-NY

We _highly_ recommend the use of the Bioconductor devel docker image described
[on the Bioconductor website](https://www.bioconductor.org/help/docker/).

If you have any questions, please get in touch with the Bioconductor core team
on the [Bioconductor Community Slack](https://bioc-community.herokuapp.com/).

### CODE REVIEW

**Comment by gladkia** [comment]
@LiNk-NY thanks in advance for having a look :).

**Comment by LiNk-NY** [comment]
Hi @gladkia 
Thanks for the PR. 
I don't see a common use case for this and the explanation is not quite accurate. 
You can still use `@return` above the `@export` tag.
Maintainers have multiple options, either to accept the warning, create a dedicated documentation page, or add a `@return` tag as above. 
FWIW Re-exports of  `%>%` are no longer needed after R `4.1.0`. 
Best,
Marcel

**Comment by gladkia** [comment]
HI @LiNk-NY,

1. Sorry for lack of the details explanation in the initial version of the PR. 

The purpose of this PR is to support the cases when one has an umbrella package with some re-exported functions. I've added an example 'testpkg3' that mimics the umbrella package (with the selected functions from 'stats' and 'utils' packages). 

I've also added some unit tests to show how the changes will affect BiocCheck() for packages with re-exports.

2.  Some links/comments related to this PR
 - there were some issues with the re-exports previously in cases of some packages that eventually were added to the Bioc: https://github.com/Bioconductor/Contributions/issues?q=reexports.Rd.  With the changes from this PR, there will be no similar issues in the future.
 - Example case of re-exports in tidyverse: https://github.com/tidyverse/dplyr/blob/29307bf661a8c52b892c28afa531b38461aababb/R/reexport-tibble.R
- a thread about the recommended documentation for the re-exported functions: https://community.rstudio.com/t/best-practices-for-documenting-re-exports/89254

**Comment by LiNk-NY** [comment]
Hi @gladkia 

It looks like they should use the package in the `Depends` field rather than trying to re-export a good number of functions. 

Given that only two packages (linked above) use this re-export mechanism it would not be too much to ask for them to add a `@return` / `@examples` tag in their documentation or create a dedicated documentation page. I would rather not make a special case for `man/reexports.Rd`. 

Best regards,
Marcel

**Comment by gladkia** [comment]
Hi @LiNk-NY ,

Thanks for the quick answer and great suggestions! Please find my comments below.
 
> It looks like they should use the package in the `Depends` field rather than trying to re-export a good number of functions.

Yes. That's a nice workaround. Unfortunately, in some cases, this solution will strongly affect performance (using `Depends` for the packages with a large number of functions). 
Thus a re-export mechanism in some cases might be a really good option, pretty performant and elegant.  You are able to select only the key exported functions within your umbrella package (e.g. like here: https://github.com/tidyverse/dplyr/blob/29307bf661a8c52b892c28afa531b38461aababb/R/reexport-tibble.R).

> Given that only two packages (linked above) use this re-export mechanism it would not be too much to ask for them to add a `@return` / `@examples` tag in their documentation or create a dedicated documentation page. I would rather not make a special case for `man/reexports.Rd`.

Yes, great suggestion. Unfortunately, it's not possible to add `@examples` tag in the re-exported function if you want to keep your documentation up-to-date with `roxygen2`. The `roxygen2` will fail in such a case. Indeed, the issue can be fixed by manually editing reexports.Rd.  But then, the maintenance of the package documentation will be much more complex and error-prone. 

Best,
Arek

**Comment by gladkia** [comment]
Hi @LiNk-NY,

Can I help in any way to move this PR forward?

Best,
Arek

File: vignettes/BiocCheck.Rmd [hunk 1]
```diff
@@ -510,6 +510,9 @@ lists the ones that don't contain runnable examples (either because
 there is no `examples` section or because its examples are tagged with
 `dontrun` or `donttest`). Runnable examples are a key part of literate
 programming and help ensure that your code does what you say it does.
+Currently, the man page for re-exported functions (`man/reexports.Rd`)
+is excluded from this check, as it's not possible to add runnable examples
+for the re-rexported functions.
```
Comment (LiNk-NY) [inline vignettes/BiocCheck.Rmd:515]: This is not accurate

**Review by LiNk-NY — CHANGES_REQUESTED** [review]

**Comment by LiNk-NY** [comment]
> Yes. That's a nice workaround. Unfortunately, in some cases, this solution will strongly affect performance (using `Depends` for the packages with a large number of functions). Thus a re-export mechanism in some cases might be a really good option, pretty performant and elegant.

Do you have metrics to show that it is more performant than using the `Depends` field? 
Also, 'elegant' is a matter of opinion. 

> Yes, great suggestion. Unfortunately, it's not possible to add `@examples` tag in the re-exported function 

This is not accurate, you can certainly add `@examples` to the roxygen block. 

Maintaining the added code is not worth the reward given that you have the options I mentioned previously. 

Best,
Marcel

**Pull request closed without merging**